File: Servidor/Tema8/ejercicios/Soluciones Ej1/ejercicio2poojava/src/Tabla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Juan
 */
public class Tabla {
  Celda[][] celdas;
  private int cantFilas;
  private int cantColumnas;
  public Tabla(int fi, int co)
  {
    this.celdas =new Celda[fi][co];
    this.cantFilas=fi;
    this.cantColumnas=co;
  }
  public void cargar(int fila,int columna,String valor,String cfue, String cfon)
  {
    this.celdas[fila][columna]=new Celda(valor,cfue,cfon);
  }
 private  String  inicioTabla()
  {
    return "<table border=1>";
  }
private String inicioFila()
  {
   return "<tr>";
  }
private String mostrar(int fi,int co)
  {
     return this.celdas[fi][co].toString()+""; 
  }
private String finFila()
  {
    return "</tr>";
  }

  private String finTabla()
  {
    return "</table>";
  }
   public String mostrarTabla()
  {
    String salida="<html><body>";
    salida+=this.inicioTabla();
    for(int f=0;f<this.cantFilas;f++)
    {
      salida+=this.inicioFila();
      for(int c=0;c<this.cantColumnas;c++)
      {
         salida+=this.mostrar(f,c);
      }
      salida+=this.finFila();
    }
    salida+=this.finTabla()+"</body></html>";
    return salida;
  }

}

File: Servidor/Tema8/ejercicios/Soluciones Ej1/ejercicio2/tabla.php
<?php 
class Tabla {
  private $mat=array();
  private $cantFilas;
  private $cantColumnas;
  public function __construct($fi,$co)
  {
    $this->cantFilas=$fi;
    $this->cantColumnas=$co;
  }
  
  public function cargar($fila,$columna,$valor,$fuente,$fondo)
  {
    $this->mat[$fila][$columna]=new Celda($valor,$fuente,$fondo);
	
  }

  private function inicioTabla()
  {
    echo '<table border="1">';
  }

  private function inicioFila()
  {
    echo '<tr>';
  }
  
  private function mostrar($fi,$co)
  {
   
	$this->mat[$fi][$co]->pintar();
  }

  private function finFila()
  {
    echo '</tr>';
  }

  private function finTabla()
  {
    echo '</table>';
  }

  public function graficar()
  {
    $this->inicioTabla();
    for($f=1;$f<=$this->cantFilas;$f++)
    {
      $this->inicioFila();
      for($c=1;$c<=$this->cantColumnas;$c++)
      {
        $this->mostrar($f,$c);
      } 
      $this->finFila();
    }
    $this->finTabla();

  }
}
?>